src/utils/httpClient.ts

```diff
@@ -164,6 +165,31 @@ export class HttpClient {
             }
         }
 
+        // set AWS authentication
+        const xAuthentication = extractHeader(options.headers!, 'X-Authentication-Type') as string | undefined;
+        if (xAuthentication) {
+            const [ xAuthenticationType, ...rawCredentials ] = xAuthentication.split(" ");
+            if (xAuthenticationType === 'AWS' ) {
+                const credentials = {
+                    accessKeyId: rawCredentials[0],
+                    secretAccessKey: rawCredentials[1],
+                    sessionToken: rawCredentials[2]
+                };
+
+                const awsScope: { [key: string]: string } = {};
+                const region = extractHeader(options.headers!, 'X-AWS-Region') as string | undefined;
+                if (region) { awsScope.region = region; }
+                const service = extractHeader(options.headers!, 'X-AWS-Service') as string | undefined;
+                if (service) { awsScope.service = service; }
+
+                options.hooks!["beforeRequest"] = [
+                    async options => {
+                        aws4.sign({ ...options, ...awsScope }, credentials);
+                    }
+                ];
+            }
+        }
+
```

Comment:
How about move this logic into a separate file under the `auth` folder like `Digest` auth?

Reply:
I've changed it to resemble the structure of Digest auth. Let me know what you think. Thanks!